windowPreview.js

```diff
@@ -362,13 +374,28 @@ class WindowPreviewMenuItem extends PopupMenu.PopupBaseMenuItem {
                                      x_expand:true });
         box.add(overlayGroup);
         box.add(labelBin);
+        this._box = box;
         this.add_actor(box);
 
         this._cloneTexture(window);
 
         this.connect('destroy', this._onDestroy.bind(this));
     }
 
+    vfunc_style_changed() {
+        super.vfunc_style_changed();
+
+        // For some crazy clutter / St reason we can't just have this handled
+        // automatically or here via vfunc_allocate + vfunc_get_preferred_*
+        // because if we do so, the St paddings on first / last child are lost
+        const themeNode = this.get_theme_node();
+        let [minWidth, naturalWidth] = this._box.get_preferred_width(-1);
+        let [minHeight, naturalHeight] = this._box.get_preferred_height(naturalWidth);
+        [minWidth, naturalWidth] = themeNode.adjust_preferred_width(minWidth, naturalWidth);
+        [minHeight, naturalHeight] = themeNode.adjust_preferred_height(minHeight, naturalHeight);
+        this.set({ minWidth, naturalWidth, minHeight, naturalHeight });
+    }
```

Comment:
@fmuellner Hey, do you have any clue why this ugly stuff is the only way to be able to get the size to be adjusted to use the actual padding to that is defined for first and last children?

To explain, this is basically just a `PopupMenu.PopupBaseMenuItem`, if from css I set something like:

```scss
       &:ltr:first-child {
            padding-left: 100px;
            padding-top: 40px;
        }
        &:ltr:last-child {
            padding-right: 100px;
        }
```
then, the children are positioned like if that is taken into account, but the size is not adjusted (in fact `vfunc_get_natural_width` is never called after the first time is set), and forcing allocating would never get the first/last items size in account.

So getting this:

![image](https://user-images.githubusercontent.com/345675/195960592-4f0ed95d-58dd-4cea-921b-d25a521bf928.png)

Instead of the expected

![image](https://user-images.githubusercontent.com/345675/195960640-72e8de2a-8bcc-4621-924b-0bf8a69e7bee.png)

Reply:
Sure, sure... Here it seems though that it's correctly computed and assigned, the problem is that the size of `PopupBaseMenuItem` is not adjusted when the first children is set to have some specific padding values.

It's weird, as applying something similar upstream works in all the other menu items I tried, while this one has nothing special.